src/backend/distributed/operations/repair_shards.c

```diff
@@ -1201,25 +1230,27 @@ CopyShardTablesViaBlockWrites(List *shardIntervalList, char *sourceNodeName,
 													 &shardForeignConstraintCommandList,
 													 &referenceTableForeignConstraintList);
 
-		List *commandList = NIL;
-		commandList = list_concat(commandList, shardForeignConstraintCommandList);
-		commandList = list_concat(commandList, referenceTableForeignConstraintList);
-
-		if (PartitionTable(shardInterval->relationId))
-		{
-			char *attachPartitionCommand =
-				GenerateAttachShardPartitionCommand(shardInterval);
-
-			commandList = lappend(commandList, attachPartitionCommand);
-		}
+		ShardIntervalWithCommandList *intervalWithCommandList = palloc0(
+			sizeof(ShardIntervalWithCommandList));
+		intervalWithCommandList->shardInterval = shardInterval;
```

Comment:
good be nice to have a CreateShardIntervalWithCommandList function.

Reply:
Added.